### PR TITLE
Run pyrsia_node as service on macOS

### LINKED ISSUE #1202: Enhance Pyrsia Installer :: Run pyrsia_node as a service at start up
### Suggestion details

Enhance Pyrsia Installer :: Run pyrsia_node as a service at start up

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by dasmanas**
## Description

Closes pyrsia/pyrsia#1202, pyrsia/pyrsia#1264
This PR takes care following items

1. test `pyrsia_node` before starting as a service using `brew test`
2. run `pyrsia_node` as a service on macOS using `brew service`.
3. takes care of the service related data (i.e. keypair, downloads etc.) under one manageable place. i.e. `#{HOMEBREW_PREFIX}/var/pyrsia`

## How to test
brew tap expects the latest is in main branch of the repo. Until and unless the PR is merge we **can't install** it using

To install use

```
brew tap pyrsia/pyrsia
brew install pyrsia
```

To test use

```
brew test pyrsia
```

To run as a service use

```
brew services start pyrsia
```

To validate use

```
brew services list
```

Pyrsia should appear in the list.

```text
Name    Status  User   File
pyrsia  started <user> ~/Library/LaunchAgents/homebrew.mxcl.pyrsia.plist
unbound none  
```

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [x] I've run the unit tests `brew test pyrsia` and everything passes.
- [x] I've made sure my brew is current and updated.

### CODE REVIEW

File: Formula/pyrsia.rb [hunk 1]
```diff
@@ -12,7 +12,57 @@ def install
     bin.install "pyrsia_node"
   end
 
+  def post_install
+    (var/"pyrsia").mkpath
```
Comment (fishseabowl) [inline Formula/pyrsia.rb:16]: Is it possible not to hardcode `var`?

File: Formula/pyrsia.rb [hunk 2]
```diff
@@ -12,7 +12,57 @@ def install
...
+
+  def caveats
+    s = <<~EOS
+      We've installed your Pyrsia Node.
+      To test Pyrsia installation:
+        brew test pyrsia
+      To run Pyrsia Node as a background service:
+        brew services start pyrsia 
+      To check the service status:
+        brew services list
```
Comment (fishseabowl) [inline Formula/pyrsia.rb:27]: It might be better to add stop/restart the Pyrsia service

File: Formula/pyrsia.rb (hunk 1 above)
Comment (dasmanas) [reply Formula/pyrsia.rb:16]: Here the `var` is defined by Homebrew. You may refer to [Variables for directory locations](https://docs.brew.sh/Formula-Cookbook#variables-for-directory-locations).

File: Formula/pyrsia.rb (hunk 2 above)
Comment (dasmanas) [reply Formula/pyrsia.rb:27]: I included the `brew services stop`.

Restart is automatically included by brew. Here is the snippet from the terminal.
```
==> Caveats
We've installed your Pyrsia Node.
To test Pyrsia installation:
  brew test pyrsia
To run Pyrsia Node as a background service:
  brew services start pyrsia 
To check the service status:
  brew services list

To restart dasmanas/pyrsiaanother/pyrsia after an upgrade:
  brew services restart dasmanas/pyrsiaanother/pyrsia
Or, if you don't want/need a background service you can just run:
  RUST_LOG="info,pyrsia=debug" /usr/local/opt/pyrsia/bin/pyrsia_node
```

**Review by ihcomega56 — APPROVED** [review]
LGTM. 
It worked, and the installer got improved definitely 👍

**Review by fishseabowl — APPROVED** [review]

**Pull request merged**